src-tauri/src/services/network_service.rs

```diff
@@ -1,17 +1,39 @@
-use std::{sync::Arc, time::Duration};
+use std::{fmt::Display, sync::Arc, time::Duration};
 
 use async_trait::async_trait;
 use tauri::async_runtime::Mutex;
 use tauri_plugin_http::reqwest;
 
 use crate::config;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Clone, Copy, Eq)]
+pub enum FrontEndEvent {
+    NetworkMode,
+}
+
+impl Display for FrontEndEvent {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FrontEndEvent::NetworkMode => write!(f, "NetworkMode"),
+        }
+    }
+}
```

Comment:
Why doesn't this impl something like Event? Display seems like a weird name.

Reply:
Display is a trait that is implemented for types that should be displayed for the user. Also, its implementation provides the ˋ.to_string()` method.

When sending events to the frontend, string must be used.

So this just add a string value for the enumeration fields.